applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_reaction_response_function.h

```diff
@@ -0,0 +1,282 @@
+                                             const ProcessInfo& rProcessInfo) override;
+
+    void CalculatePartialSensitivity(Element& rAdjointElement,
+                                             const Variable<array_1d<double, 3>>& rVariable,
+                                             const Matrix& rSensitivityMatrix,
+                                             Vector& rSensitivityGradient,
+                                             const ProcessInfo& rProcessInfo) override;
+
+    void CalculatePartialSensitivity(Condition& rAdjointCondition,
+                                             const Variable<array_1d<double, 3>>& rVariable,
+                                             const Matrix& rSensitivityMatrix,
+                                             Vector& rSensitivityGradient,
+                                             const ProcessInfo& rProcessInfo) override;
+
+
+    double CalculateValue(ModelPart& rModelPart) override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+
+    ///@}
+
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+
+    ///@}
+
+private:
+
+    ///@name Static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Member Variables
+    ///@{
+
+    std::string mTracedDisplacementDofLabel;
+    std::string mTracedReactionLabel;
+    PointTypePointer  mpTracedNode;
+    WeakPointerVector<Element> mpNeighborElements;
+    WeakPointerVector<Condition> mpNeighborConditions;
+    bool mAdjustInfluenceFunction = false;
+    bool mPerformedModificationOfAdjointDisplacements = false;
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+
+    void CalculateContributionToPartialSensitivity(Element& rAdjointElement,
+                                             const Matrix& rSensitivityMatrix,
+                                             Vector& rSensitivityGradient,
+                                             const ProcessInfo& rProcessInfo);
+
+    void CalculateContributionToPartialSensitivity(Condition& rAdjointCondition,
+                                             const Matrix& rSensitivityMatrix,
+                                             Vector& rSensitivityGradient,
+                                             const ProcessInfo& rProcessInfo);
+
+    template <typename TObjectType>
+    void ConstructFilterVector(TObjectType& rAdjointObject, Vector& rFilterVector, ProcessInfo& rProcessInfo)
```

Comment:
documentation

Reply:
Do i understand it correctly that you are using a matrix vector multiplication with a filter vector [0,0,-1,0,0] to get a specific negative row/column?
I have the feeling that there has to be a better/easier way to achieve this.

Reply:
This is what I am doing. I am not sure if there is a functionality the get the needed data. I will have a look.

Reply:
In principle a Matrix has functionalities the get a column or row. In case of `CalculateGradient`this perfectly works but in `CalculatePartialSensitivity` the sensitivity matrix is given as `const` which forbids the usage of these functions. In this case a local copy of the sensitivity matrix would be needed what is also not nice.

Reply:
Ok. Then i still think a function like `Vector GetRowCopy(Matrix m, size_t i)` in combination with `size_t GetDofIndex(dof)` and then multiplying the vector with -1 fits better to what you are actually doing.

Reply:
+1 for `GetRowCopy`. AMatrix probably needs a `const row(int)` function which returns a const view.

Reply:
wait there is no `const` version of `row`?